evennia/contrib/mapbuilder.py

```diff
@@ -0,0 +1,458 @@
+    room_dict - A dictionary containing room references returned by build
+                functions where tuple coordinates are the keys (x, y).
+                ie room_dict[(2, 2)] will return the temple room above.
+
+Building functions should return the room they create. By default these rooms
+are used to create exits between valid adjacent rooms to the north, south,
+east and west directions. This behaviour can turned off with the use of switch
+arguments. In addition to turning off automatic exit generation the switches
+allow the map to be iterated over a number of times. This is important for
+something like custom exit building. Exits require a reference to both the
+exits location and the exits destination. During the first iteration it is
+possible that an exit is created pointing towards a destination that
+has not yet been created resulting in error. By iterating over the map twice
+the rooms can be created on the first iteration and room reliant code can be
+be used on the second iteration. The iteration number and a dictionary of
+references to rooms previously created is passed to the build commands.
+
+Use by importing and including the command in your default_cmdsets module.
+For example:
+
+    # mygame/commands/default_cmdsets.py
+
+    from evennia.contrib import mapbuilder
+
+    ...
+
+    self.add(mapbuilder.CmdMapBuilder())
+
+You then call the command in-game using the path to the MAP and MAP_LEGEND vars
+The path you provide is relative to the evennia or mygame folder.
+
+Usage:
+    @mapbuilder[/switch] <path.to.file.MAPNAME> <path.to.file.MAP_LEGEND>
+
+Switches:
+  one - execute build instructions once without automatic exit creation.
+  two - execute build instructions twice without automatic exit creation.
+
+Example:
+    @mapbuilder world.gamemap.MAP world.maplegend.MAP_LEGEND
+    @mapbuilder evennia.contrib.mapbuilder.EXAMPLE2_MAP EXAMPLE2_LEGEND
+            (Legend path defaults to map path)
+
+Below are two examples showcasing the use of automatic exit generation and
+custom exit generation. Whilst located, and can be used, from this module for
+convenience The below example code should be in mymap.py in mygame/world.
+
+"""
+
+from django.conf import settings
+from evennia.utils import utils
+
+# ---------- EXAMPLE 1 ---------- #
+# @mapbuilder evennia.contrib.mapbuilder.EXAMPLE1_MAP EXAMPLE1_LEGEND
+
+# -*- coding: utf-8 -*-
+
+# Add the necessary imports for your instructions here.
+from evennia import create_object
+from typeclasses import rooms, exits
+from evennia.utils import utils
+from random import randint
+import random
+
+# A map with a temple (▲) amongst mountains (n,∩) in a forest (♣,♠) on an
+# island surrounded by water (≈). By giving no instructions for the water
+# characters we effectively skip it and create no rooms for those squares.
+EXAMPLE1_MAP = """\
+≈≈≈≈≈
+≈♣n♣≈
+≈∩▲∩≈
+≈♠n♠≈
+≈≈≈≈≈
+"""
+
+
+def build_forest(x, y, **kwargs):
```

Comment:
I don't think this function will ever be used, since there is a same-named function below that will replace it.